modules/misc.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+# Copyright 2022 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+function merge_mender_configs() {
+    declare -a ol_targets=()
+    local log="work/mergefiles/log"
+    local tmp="work/mergefiles/tmp"
+
+    for overlay in "${overlays[@]}"; do
+        [ -f "${overlay}/etc/mender/mender.conf" ] && ol_targets+=("${overlay}")
+    done
+
+    mkdir -p work/mergefiles
+    local -r menderconfs="input/resources/mender.conf ${ol_targets[@]/%//etc/mender/mender.conf}"
+
+    if [ "${#ol_targets[@]}" -ne 0 ]; then
+        echo "Attempting to merge /etc/mender/mender.conf in overlays. With respect to the overlays, the merge is done in the following order (first used as base):" >> "${log}"
+        echo -n " * input/resources (generated by the bootstrap script)" >> "${log}"
+        echo -e "${ol_targets[@]/#/'\n * '}" >> "${log}"
+        log_info "$(cat ${log})"
+        run_and_log_cmd "jq --slurp ""'reduce .[] as "'$item ({}; . '"'*'"' $item)'"'"" ${menderconfs} > ${tmp}"
```

Comment:
This line is a bit ridiculous with all the nested quoting/concatenations. However `run_and_log_cmd` isn't really good with specials chars. I'm considering changing `run_and_log_cmd` to make cases like these more bearable.

Reply:
haha, interesting. This we should probably have a think about :)

Reply:
At least we should add a comment in the code on why this command is so weird

Reply:
Discussed f2f, will do a command file here for increased prettiness :rainbow: :unicorn: